kore/src/Kore/Internal/NormalizedAc.hs

```diff
@@ -509,22 +513,48 @@ normalizedAcConstructorLike ac@(NormalizedAc _ _ _) =
                     assertConstructorLike "" key $ isConstructorLike value
         _ -> ConstructorLike Nothing
 
+{- | A representation for associative-commutative collections
+with just two elements of the same type.
+
+The smart constructor 'mkAcPair' ensures the properties of
+the collection are satisfied.
+-}
+data AcPair a = AcPair_ a a
+    deriving stock (Eq)
+    deriving stock (GHC.Generic)
+    deriving anyclass (Hashable)
+
+pattern AcPair :: a -> a -> AcPair a
```

Comment:
Haven't we been using a convention where the constructor is given the regular name and the pattern is given the name with an underscore?

Reply:
We only use the convention that pattern synonyms have a trailing `_` in `Kore.Internal.TermLike`. (Those constructors are also exported.) Elsewhere, e.g. `Kore.Internal.Substitution`, we use the same convention that is used here.